sdk/core/Azure.Core.Experimental/src/Variant/Variant.cs

```diff
@@ -22,8 +22,16 @@ namespace Azure
         /// <param name="value"></param>
         public Variant(object? value)
         {
-            _object = value;
-            _union = default;
+            if (value is Variant variant)
+            {
+                _object = variant._object;
```

Comment:
Would it make sense to clone the object here rather than using the same instance?

Reply:
Ah, very good question.  Let me think more about what we want here.  Thank you!

Reply:
I think we want Variant to behave like object when it is holding a reference type.  Since object uses reference semantics, I feel that Variant should do the same.  e.g. for object, we would have:

```csharp
    List<string> list = new List<string> { "1" };
    object oa = list;
    object ob = oa;

    Assert.AreEqual("1", list[0]);
    Assert.AreEqual("1", ((List<string>)oa)[0]);
    Assert.AreEqual("1", ((List<string>)ob)[0]);

    list[0] = "2";
  
    Assert.AreEqual("2", list[0]);
    Assert.AreEqual("2", ((List<string>)oa)[0]);
    Assert.AreEqual("2", ((List<string>)ob)[0]);
```

I'll add tests to illustrate this and validate the functionaliry.

Reply:
I don't know if the above test is equivalent to what we are doing here. Shouldn't it be:

```c#
List<string> list = new List<string> { "1" };
object oa = new List<string>(list);
object ob = new List<string>((List<string>)oa);
```

The second set of asserts fail in this case.

Reply:
In your example, you are creating a copy of `list` and assigning it to `oa`.  In my example, I am assigning `oa` to `list`.  Setting a `Variant` instance to a reference type is the equivalent of the latter, not the former.  I don't think we want to clone any reference type to assign it to Variant -- that would make it high performance for value types but not reference types, which feels counter to the purpose of the type to me.

Reply:
I thought this was for the scenario where the Variant constructor is called with an object, which would be equivalent to my example.

Reply:
There are some reference types that you can assign to Variant without calling the Variant constructor, like string.  However, in the general case, calling the Variant constructor with an object is how you create an instance of Variant that holds that object.  So, I believe the equivalent of your example with Variant would be:

```csharp
List<string> list = new List<string> { "1" };
Variant va = new Variant(new List<string>(list));
Variant vb = new Variant(new List<string>(oa.As<List<string>>);
```

But, do you have evidence to the contrary?  If I have a bug in my brain, I would like to fix it!

Reply:
> Would it make sense to clone the object here rather than using the same instance?

How would we clone it? there is not a reliable way to clone an arbitrary .NET object.

Reply:
> > Would it make sense to clone the object here rather than using the same instance?
> 
> How would we clone it? there is not a reliable way to clone an arbitrary .NET object.

Good question - we can't clone it for arbitrary types.